src/main/java/me/lemire/integercompression/SkippableIntegerCODEC.java

```diff
@@ -10,7 +10,7 @@
 
 /**
  * Interface describing a standard CODEC to compress integers. This is a
- * variation on the IntegerCODEC interface meant to be used for random access.
+ * variation on the IntegerCODEC interface meant to be used for head access.
```

Comment:
Fixed in the main branch.

Reply:
Can you explain what we mean by random access then ? I see how we can fetch only the first N entries, but I do not see how we can skip the first M (expect by decompressing them (i.e. not a random access in my understanding)).

Reply:
If you look at the generic compress and uncompress routines, they are meant to compress and uncompress *whole* arrays entirely. They store the metadata (number of elements) within the stream.

If you only need to access some elements in a large array, it is not great. So the typical approach in a search engine is to take your large array and divide it up into small chunks/blocks, and to have a some kind of data structure on top. Then you query data from a given block and recover the elements, specifying how many elements you need.

 You, the user of the library, must provide the metadata and the organisation. The library really just allows you to compress and decompress small blocks in a sensible manner.

Reply:
Hence, SkippableIntegerCODEC enables building a random access by compressing by chunks. But this is also true for IntegerCODEC (except SkippableIntegerCODEC may enable a smaller footprint, by storing the length somewhere else). As it stands, this was confusive to myself (as a User of the library). Just the same, it took time to understand `SkippableXXX` relates to the ability to skip the rest of a head-operation (instead of skipping the head, which is to what random-access generally refers to in my understanding).

Reply:
>  SkippableIntegerCODEC enables building a random access by compressing by chunks. But this is also true for IntegerCODEC (except SkippableIntegerCODEC may enable a smaller footprint, by storing the length somewhere else).

If 1000 integers were compressed with IntegerCODEC, then you have to uncompress 1000 integers. The headless approach allows you to specify how many you need/want.

You are correct, you could use IntegerCODEC to store chunks and just uncompress the chunks you need, on demand... however, this will still leave you uncompress *whole* chunks each time. 

> As it stands, this was confusive to myself 

I agree. It could be clearer. It is documented as follows, currently...

`JavaFastPFOR supports compressing and uncompressing data in chunks (e.g., see advancedExample in https://github.com/lemire/JavaFastPFOR/blob/master/example.java).`

Reply:
JavaFastPFOR enables random-access by compressing chunks. SkippableIntegerCODEC enables head-access (or whatever the wording), but saying SkippableIntegerCODEC enables random-access is confusive to me. (Anyway, I proposed a changed, you discarded it. I'm totally fine with this process :).)

Reply:
> Anyway, I proposed a changed, you discarded it. I'm totally fine with this process :).

That's not my view. I reverted it because I was not convinced we had a good understanding of the issue. I am not sure for example that 'head access' is clearer, at least it is not clearer to me.

I do invite a revision of this documentation. It can be greatly improved.

Reply:
let me make some proposition in a different PR.